application/src/main/java/org/togetherjava/tjbot/commands/moderation/ReportCommand.java

```diff
@@ -0,0 +1,224 @@
+            .addActionRow(body)
+            .build();
+
+        event.replyModal(reportModal).queue();
+    }
+
+    private boolean handleIsOnCooldown(MessageContextInteractionEvent event) {
+        if (!isAuthorOnCooldown(event.getUser().getIdLong())) {
+            return false;
+        }
+        event
+            .reply("You can only report a message once per %s minutes."
+                .formatted(COOLDOWN_DURATION_VALUE))
+            .setEphemeral(true)
+            .queue();
+        return true;
+    }
+
+    private boolean isAuthorOnCooldown(long userId) {
+        return Optional.ofNullable(authorToLastReportInvocation.getIfPresent(userId))
+            .map(sinceCommandInvoked -> sinceCommandInvoked.plus(COOLDOWN_DURATION_VALUE,
+                    COOLDOWN_DURATION_UNIT))
+            .filter(Instant.now()::isBefore)
+            .isPresent();
+    }
+
+    @Override
+    public void onModalSubmitted(ModalInteractionEvent event, List<String> args) {
+        Optional<TextChannel> modMailAuditLog = handleRequireModMailChannel(event);
+
+        if (modMailAuditLog.isEmpty()) {
+            return;
+        }
+
+        sendModMessage(event, args, modMailAuditLog.orElseThrow());
+    }
+
+    private Optional<TextChannel> handleRequireModMailChannel(ModalInteractionEvent event) {
+        long guildID = Objects.requireNonNull(event.getGuild(), "Could not retrieve the guildId.")
+            .getIdLong();
+        Optional<TextChannel> modMailAuditLog = event.getJDA()
+            .getTextChannelCache()
+            .stream()
+            .filter(channel -> modMailChannelNamePredicate.test(channel.getName()))
+            .findAny();
+        if (modMailAuditLog.isEmpty()) {
+            event
+                .reply("Sorry, there was an issue sending your report to the moderators. We are "
+                        + "investigating.")
+                .setEphemeral(true)
+                .queue();
+            logger.warn(
+                    "Cannot find the designated modmail channel in server by id {} with the pattern {}",
+                    guildID, configModMailChannelPattern);
+        }
+        return modMailAuditLog;
+    }
+
+    private MessageCreateAction createModMessage(String reportReason,
+            ReportedMessage reportedMessage, Guild guild, TextChannel modMailAuditLog) {
+
+        MessageEmbed reportedMessageEmbed = new EmbedBuilder().setTitle("Report")
+            .setDescription(MessageUtils.abbreviate(reportedMessage.content,
+                    MessageEmbed.DESCRIPTION_MAX_LENGTH))
+            .setAuthor(reportedMessage.authorName, null, reportedMessage.authorAvatarUrl)
+            .setTimestamp(reportedMessage.timestamp)
+            .setColor(AMBIENT_COLOR)
+            .build();
+
+        MessageEmbed reportReasonEmbed = new EmbedBuilder().setTitle("Reason")
+            .setDescription(reportReason)
+            .setColor(AMBIENT_COLOR)
+            .build();
```

Comment:
can u put the reason as a field in the main embed instead of creating a new embed, it'll look better

Reply:
disagree, i asked for this specifically.